app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,47 @@
+class FriendshipsController < ApplicationController
+  def send_request
+    if current_user.send_invitation(params[:user_id])
+      flash.notice = 'Friend request sent'
+      redirect_to users_path
+    else
+      flash.now[:notice] = 'error occured'
+    end
+  end
+
+  def accept_invitation
+    if current_user.confirm_friend(params[:user_id])
+      flash.notice = 'friend accepted'
+      redirect_to users_path
+    else
+      flash.now[:notice] = 'error occured'
+    end
+  end
+
+  def reject_invitation
+    current_user.reject_friend(params[:user_id])
+    redirect_to users_path
+  end
+
+  def pending_invitation
+    @pending_friends = current_user.pending_invites
+  end
```

Comment:
**Optional** if you have `current_user` as a global variable, you can access it directly in the view and use the suggested associations instead of creating a method for it. This way you avoid unnecessary code.